src/readfish/_utils.py

```diff
@@ -449,6 +453,48 @@ def get_device(
             return position
     raise ValueError(f"Could not find device {device!r}")
 
+def get_version(
+        host: str = "127.0.0.1", port: int = None, logger: logging.Logger = None
+) -> str:
+    """Get the version of MinKNOW
+
+    :param host: The host the RPC is listening on, defaults to "127.0.0.1"
+    :param port: The port the RPC is listening on, defaults to None
+    :raises ValueError: If the version of Readfish is incompatible with the version of MinKNOW.
+    :raises Warning: If the version of Readfish may need to be updated soon.
+    :return: The version of MinKNOW
+    """
+    manager = Manager(host=host, port=port)
+    minknow_version = parse_version(manager.core_version)
+    # Check compatability
+    for cv, direction in compatible_version:
+        ver_cv = parse_version(cv)
+        if direction == "gte":
+            if minknow_version <= ver_cv:
+                raise ValueError(
+                    f"Compatability check failed. Readfish version {readfish_version} is not compatible with MinKNOW version {minknow_version}. You need to update readfish."
+                )
+        elif direction == "lte":
+            if minknow_version >= ver_cv:
+                raise ValueError(
+                    f"Compatability check failed. Readfish version {readfish_version} is not compatible with MinKNOW version {minknow_version}. You need to downgrade readfish."
+                )
+    for wv, message in warning_version:
+        warn = False
+        ver_wv = parse_version(wv)
+        if ver_wv.major == minknow_version.major:
+            if ver_wv.minor == minknow_version.minor:
+                if ver_wv.micro == minknow_version.micro:
+                    warn=True
+                else:
+                    warn=True
+            else:
+                warn=True
+        if warn:
+            logger.info(
+                f"Warning: {message}"
+            )
+    return None
```

Comment:
looking at this it could be simplified into one conditional
```python
    warn = (ver_wv.major == minknow_version.major and 
            (ver_wv.minor == minknow_version.minor or
             ver_wv.micro == minknow_version.micro))
   if warn:
        ....
```


also I'm confused what are these checks doing?? Are we saying if you run this version of readfish on this version of MinKNOW then you have to upgrade?
But the warning version is `5.9`?